hypershift-operator/controllers/nodepool/kubevirt/kubevirt.go

```diff
@@ -445,3 +450,88 @@ func applyJsonPatch(tmplt *[]byte, patch string) error {
 
 	return nil
 }
+
+func ReconcileDefaultIngressEndpoints(ctx context.Context, cli client.Client, hc *hyperv1.HostedCluster, machines []*capiv1.Machine) error {
+	var namespace string
+	if hc.Spec.Platform.Kubevirt.Credentials != nil {
+		namespace = hc.Spec.Platform.Kubevirt.Credentials.InfraNamespace
+	} else {
+		namespace = hc.Namespace + "-" + hc.Name
+	}
+
+	// Manifests for infra/mgmt cluster passthrough service
+	cpService := manifests.IngressDefaultIngressPassthroughService(namespace)
+
+	cpService.Name = fmt.Sprintf("%s-%s",
+		manifests.IngressDefaultIngressPassthroughServiceName,
+		hc.Spec.Platform.Kubevirt.GenerateID)
+
+	if err := cli.Get(ctx, client.ObjectKeyFromObject(cpService), cpService); err != nil {
+		return fmt.Errorf("failed to get default ingress passthrow Service: %w", err)
+	}
+
+	// If there is a selector endpoints should not be generated
+	if len(cpService.Spec.Selector) > 0 {
+		return nil
+	}
+
+	cpEndpoints := &corev1.Endpoints{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: cpService.Namespace,
+			Name:      cpService.Name,
+		},
+	}
+
+	if len(machines) == 0 {
+		return cli.Delete(ctx, cpEndpoints)
+	}
+
+	machineInternalIPs := []string{}
+	for _, machine := range machines {
+		machineInternalIP := findMachineInternalIP(machine)
+		if machineInternalIP != "" {
+			machineInternalIPs = append(machineInternalIPs, machineInternalIP)
+		}
+	}
+
+	endpointSubsetAddresses := []corev1.EndpointAddress{}
+	for _, machineInternalIP := range machineInternalIPs {
+		endpointSubsetAddresses = append(endpointSubsetAddresses, corev1.EndpointAddress{
+			IP: machineInternalIP,
+		})
+	}
+	if len(cpService.Spec.Ports) == 0 {
+		return fmt.Errorf("missing default ingress passthrow Service %s/%s ports", cpService.Namespace, cpService.Name)
+	}
+	endpointSubsetPorts := []corev1.EndpointPort{}
+	for _, port := range cpService.Spec.Ports {
+		endpointSubsetPorts = append(endpointSubsetPorts, corev1.EndpointPort{
+			Port: port.TargetPort.IntVal,
+		})
+	}
+
+	err := cli.Get(ctx, client.ObjectKeyFromObject(cpEndpoints), cpEndpoints)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("failed to get default ingress passthrow Endpoints: %w", err)
+	}
+
+	cpEndpoints.Subsets = []corev1.EndpointSubset{{
+		Addresses: endpointSubsetAddresses,
+		Ports:     endpointSubsetPorts,
+	}}
```

Comment:
I believe this is only accurate if a hcp has a single NodePool associated with it. If the hcp has multiple nodepools (which is definitely a use case) then each nodepool will override the endpoint subsets. 

There are a couple of options here to fix this.
1. you could use the `EndpointAddress.TargetRef` object to ensure that each nodepool reconcile is only modifying the endpoints associated with its nodepool. This might lead to some tricky cases during NodePool deletion though.
2. You could move all this logic to a controller that has a cluster wide view of all the nodes.

Reply:
We can list all the machines and reconcile, is not optimal but is ok.

Reply:
We can do this part of the reconcile for all the cluster's machine, since we don't really care for nodepool, and the conflict error will trigger Reconcile cycle again if two controllers update the service at the same time, hitting too much the api server is not a problem since the actual code is already listing all the machines at the namespace and filtering by annotation.

About deletion, I will remove the endpoint deletion feature, having it there without Subsets is better than breaking the feature and it has already a owner so it will be removed.